custom-routes-vis/src/main/webapp/script.js

```diff
@@ -201,77 +205,82 @@ function generateRoutes() {
   let routeCoordinates = [];
 
   // Total duration of route in seconds.
-  let totalDuration = directionsApi ? 0 : routeJson.duration;
+  let totalRouteDurationSec = directionsApi ? 0 : routeJson.duration;
   // Total distance of route in meters.
-  let totalDistance = directionsApi ? 0 : routeJson.distanceMeters;
+  let totalDistanceMeters = directionsApi ? 0 : routeJson.distanceMeters;
   // Note: Routes Preferred has duration and distanceMeters attributes for each
-  // route, but Directions only has those for each leg of the route.
+  // route, but Directions only has them for each leg of the route.
 
+  // Get all coordinates given by the steps of the legs of the route.
   for (let i = 0; i < routeLegs.length; i++) {
     let legSteps = routeLegs[i].steps;
     console.log('num steps', legSteps.length);
 
     for (let j = 0; j < legSteps.length; j++) {
-      if (directionsApi) {
-        routeCoordinates.push(legSteps[j].start_location);
-      } else {
-        routeCoordinates.push(legSteps[j].startLocation.LatLng);
-      }
+      directionsApi ?
+          routeCoordinates.push(legSteps[j].start_location) :
+          routeCoordinates.push(legSteps[j].startLocation.LatLng);
     }
 
     // Add the end location to the coordinates array.
-    if (i == routeLegs.length - 1) {
-      if (directionsApi) {
-        routeCoordinates.push(legSteps[legSteps.length - 1].end_location);
-      } else {
-        routeCoordinates.push(legSteps[legSteps.length - 1].endLocation.LatLng);
-      }
+    if (i === routeLegs.length - 1) {
+      directionsApi ?
+          routeCoordinates.push(legSteps[legSteps.length - 1].end_location) :
+          routeCoordinates.push(legSteps[legSteps.length - 1].endLocation.LatLng);
     }
 
     if (directionsApi) {
-      totalDuration += parseInt(routeLegs[i].duration.value);
-      totalDistance += parseInt(routeLegs[i].distance.value);
+      // Accumulate duration and distance because the total is not directly
+      // available in results from the Directions API.
+      totalRouteDurationSec += parseInt(routeLegs[i].duration.value);
+      totalDistanceMeters += parseInt(routeLegs[i].distance.value);
     }
   }
 
-  createRouteFromCoordinates(routeCoordinates, routeNum, totalDuration,
-      totalDistance);
+  createRouteFromCoordinates(routeCoordinates, routeNum, totalRouteDurationSec,
+      totalDistanceMeters);
 }
 
 /**
  * Creates the route on the map and stores it in the routes array.
+ * @param {Array} routeCoordinates The coordinates that the route goes through.
+ * @param {num} routeNum The index of the selected route in the routes array.
+ * @param {num} totalDuration The duration of the route in seconds.
+ * @param {num} totalDistance The distance of the route in meters.
  */
-function createRouteFromCoordinates(routeCoordinates, routeNum, totalDuration, 
+function createRouteFromCoordinates(
+    routeCoordinates, 
+    routeNum, 
+    totalDuration, 
     totalDistance) {
```

Comment:
I think you probably want to rename this to have similar naming conventions as you do above? So totalDurationSec etc.